### PR TITLE
feat(navbar-brand): New component <b-navbar-brand>

### PR DESCRIPTION
**Pull request by tmorehouse**
New component for rendering navbar brand content.

Setting props `to` or `href` will render a `b-link`. 

If both `to` and `href` are not present, then the navbar-brand will be resndered as a `div` by default (configurable via the `tag` prop)

### CODE REVIEW

File: lib/components/navbar-brand.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<template>
+     <b-link v-if="isLink"
+             class="navbar-brand"
+             v-bind="linkProps"
+             @click="$emit('click', $event)">
+        <slot></slot>
+    </b-link>
+    <div v-else :is="tag" class="navbar-brand">
+        <slot></slot>
+    </div>
```
Comment (alexsasharegan) [inline lib/components/navbar-brand.vue]: Might make slightly more sense just using the `<component>` tag here. Just shows it off clearly as a dynamic tag.

File: lib/components/navbar-brand.vue [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<template>
+     <b-link v-if="isLink"
+             class="navbar-brand"
+             v-bind="linkProps"
+             @click="$emit('click', $event)">
```
Comment (alexsasharegan) [inline lib/components/navbar-brand.vue]: Once we hit v2.4 as a peer dependency with Vue, we can start using `v-bind="$attrs" v-on="$listeners"` to proxy everything off to link or whatever. And that is glorious!

File: lib/components/navbar-brand.vue (hunk 1 above)
Comment (tmorehouse) [inline lib/components/navbar-brand.vue]: Yeah, was thinking that

File: lib/components/navbar-brand.vue (hunk 2 above)
Comment (tmorehouse) [inline lib/components/navbar-brand.vue]: Excellent. Will be nice when that happens

**Pull request merged**